### PR TITLE
Add datastore flutter code snippets

### PR DESCRIPTION
**Pull request by Amplifiyer**
This PR doesn't add all the snippets, still TBDs are

* Relational data access
* Sync
* Events
* Auth rules
* Conflict resolution

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/lib/analytics/fragments/flutter/getting-started/10_preReq.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
 * [Install and configure Amplify CLI](https://docs.amplify.aws/cli/start/install)
 * A Flutter application targeting Flutter SDK >= 1.20 with Amplify libraries integrated
     * An iOS configuration targeting at least iOS 11.0
-    * An Android configuration targeting at least Android API level 16 (Android 4.1) or above
+    * An Android configuration targeting at least Android API level 21 (Android 5.0) or above
```
Comment (haverchuck) [inline docs/lib/analytics/fragments/flutter/getting-started/10_preReq.md:4]: Thanks for remembering these!

File: docs/lib/datastore/fragments/flutter/data-access/delete-snippet.md [hunk 2]
```diff
@@ -1,5 +1,13 @@
-To delete an item, simply pass in an instance to `Amplify.DataStore.delete()`.  Below, we query for an instance with an `id` of `"123"`, and then delete it, if found:
```
Comment (haverchuck) [inline docs/lib/datastore/fragments/flutter/data-access/delete-snippet.md:1]: Why delete this?  It is not STRICTLY necessary to query for an instance first, so this first sentence adds a bit of clarity.

File: docs/lib/datastore/fragments/flutter/data-access/delete-snippet.md [hunk 3]
```diff
@@ -1,5 +1,13 @@
...
+Below, we query for an instance with an `id` of `"123"`, and then delete it, if found:
 
 ```dart
-TBD
-```
\ No newline at end of file
+(await Amplify.DataStore.query(Post.classType, where: Post.ID.eq("123")))
```
Comment (haverchuck) [inline docs/lib/datastore/fragments/flutter/data-access/delete-snippet.md:4]: Thank you for taking care of the delete documentation.

**Review by haverchuck — COMMENTED** [review]
Please see in-line comments.  Other than those minor comments, LGTM.

File: docs/lib/datastore/fragments/flutter/data-access/delete-snippet.md (hunk 2 above)
Comment (Amplifiyer) [reply docs/lib/datastore/fragments/flutter/data-access/delete-snippet.md:1]: This first sentence was repeated twice, at least when running in my local instance that's why I removed it, totally agree that it's important.

File: docs/lib/datastore/fragments/flutter/getting-started/70_querySnippet.md [hunk 4]
```diff
@@ -1,3 +1,7 @@
 ```dart
-TBD
-```
\ No newline at end of file
+try {
+  List<Post> posts = query(Post.classType);
```
Comment (jamesonwilliams) [inline docs/lib/datastore/fragments/flutter/getting-started/70_querySnippet.md:3]: Does it not need to be:
```suggestion
  List<Post> posts = await Amplify.DataStore.query(Post.classType);
```

File: docs/lib/datastore/fragments/flutter/data-access/query-basic-snippet.md [hunk 5]
```diff
@@ -1,3 +1,7 @@
 ```dart
-TBD
-```
\ No newline at end of file
+try {
+  List<Post> posts = query(Post.classType);
```
Comment (jamesonwilliams) [inline docs/lib/datastore/fragments/flutter/data-access/query-basic-snippet.md:3]: Does it not need to be:
```suggestion
  List<Post> posts = await Amplify.DataStore.query(Post.classType);
```

File: docs/lib/datastore/fragments/flutter/getting-started/70_querySnippet.md (hunk 4 above)
Comment (Amplifiyer) [reply docs/lib/datastore/fragments/flutter/getting-started/70_querySnippet.md:3]: You are right, bad copy paste error. Thanks for catching.

**Review by jamesonwilliams — APPROVED** [review]

**Pull request merged**